File: mycollab-web/src/main/scala/com/mycollab/module/crm/httpmapping/OpportunityUrlResolver.scala
/**
 * This file is part of mycollab-web.
 *
 * mycollab-web is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-web is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-web.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.module.crm.httpmapping

import com.mycollab.common.UrlTokenizer
import com.mycollab.eventmanager.EventBusFactory
import com.mycollab.module.crm.domain.Account
import com.mycollab.module.crm.event.OpportunityEvent

/**
  * @author MyCollab Ltd
  * @since 5.0.9
  */
class OpportunityUrlResolver extends CrmUrlResolver {
  this.addSubResolver("list", new OpportunityListUrlResolver)
  this.addSubResolver("add", new OpportunityAddUrlResolver)
  this.addSubResolver("edit", new OpportunityEditUrlResolver)
  this.addSubResolver("preview", new OpportunityPreviewUrlResolver)

  class OpportunityListUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      EventBusFactory.getInstance().post(new OpportunityEvent.GotoList(this, null))
    }
  }

  class OpportunityAddUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      EventBusFactory.getInstance().post(new OpportunityEvent.GotoAdd(this, new Account))
    }
  }

  class OpportunityEditUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      val opportunityId = UrlTokenizer(params(0)).getInt
      EventBusFactory.getInstance().post(new OpportunityEvent.GotoEdit(this, opportunityId))
    }
  }

  class OpportunityPreviewUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      val opportunityId = UrlTokenizer(params(0)).getInt
      EventBusFactory.getInstance().post(new OpportunityEvent.GotoRead(this, opportunityId))
    }
  }

}

File: mycollab-web/src/main/java/com/mycollab/module/crm/view/opportunity/OpportunityUrlResolver.kt
/**
 * mycollab-web - Parent pom providing dependency and plugin management for applications
		built with Maven
 * Copyright © ${project.inceptionYear} MyCollab (support@mycollab.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.module.crm.view.opportunity

import com.mycollab.common.UrlTokenizer
import com.mycollab.vaadin.EventBusFactory
import com.mycollab.module.crm.domain.Account
import com.mycollab.module.crm.event.OpportunityEvent
import com.mycollab.module.crm.view.CrmUrlResolver

class OpportunityUrlResolver : CrmUrlResolver() {
    init {
        this.addSubResolver("list", OpportunityListUrlResolver())
        this.addSubResolver("add", OpportunityAddUrlResolver())
        this.addSubResolver("edit", OpportunityEditUrlResolver())
        this.addSubResolver("preview", OpportunityPreviewUrlResolver())
    }

    class OpportunityListUrlResolver : CrmUrlResolver() {
        override fun handlePage(vararg params: String) =
                EventBusFactory.getInstance().post(OpportunityEvent.GotoList(this, null))

    }

    class OpportunityAddUrlResolver : CrmUrlResolver() {
        override fun handlePage(vararg params: String) =
                EventBusFactory.getInstance().post(OpportunityEvent.GotoAdd(this, Account()))
    }

    class OpportunityEditUrlResolver : CrmUrlResolver() {
        override fun handlePage(vararg params: String) {
            val opportunityId = UrlTokenizer(params[0]).getInt()
            EventBusFactory.getInstance().post(OpportunityEvent.GotoEdit(this, opportunityId))
        }
    }

    class OpportunityPreviewUrlResolver : CrmUrlResolver() {
        override fun handlePage(vararg params: String) {
            val opportunityId = UrlTokenizer(params[0]).getInt()
            EventBusFactory.getInstance().post(OpportunityEvent.GotoRead(this, opportunityId))
        }
    }
}